locations/locations.go

```diff
@@ -27,9 +27,14 @@ import (
 	"github.com/jhump/protoreflect/desc"
 )
 
-// pathLocation returns the precise location within a file for a given path.
-func pathLocation(f *desc.FileDescriptor, path []int32) *dpb.SourceCodeInfo_Location {
-	return sourceInfoRegistry.sourceInfo(f).findLocation(path)
+// pathLocation returns the precise location for a given descriptor and path.
+// It combines the path of the descriptor itself with any path provided appended.
+func pathLocation(d desc.Descriptor, path ...int) *dpb.SourceCodeInfo_Location {
+	fullPath := d.GetSourceInfo().GetPath()
```

Comment:
Check if `d.GetSourceInfo() == nil`

Reply:
Believe it or not, unnecessary! :-) All of the `FileLocation` tests (`FileDescriptor` always returns nil) pass and do the correct thing.